src/http/Client.js

```diff
@@ -127,7 +128,13 @@ export default class Client extends EventEmitter {
 
             // Assign a new encoded body
             if (contentType.indexOf(ApiResponse._jsonContentType) > -1) {
-                init.body = JSON.stringify(init.body);
+                if((init.method === 'GET' || init.method === 'HEAD') && !!init.body) {
+                    // oddly setting body to null still result in TypeError in phantomjs
+                    init.body = void 0;
```

Comment:
hmm... just set it to undefined I guess. Right, we shouldn't have to worry about undefined being overridden anymore. I'll push another change then.